sdk/client/src/blocking.rs

```diff
@@ -1,31 +1,104 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
+use std::time::Duration;
+
 use super::{
     request::{JsonRpcRequest, MethodRequest},
     response::{MethodResponse, Response},
+    state::StateManager,
     validate, validate_batch, BatchResponse,
 };
 use crate::{
+    error::WaitForTransactionError,
     views::{
         AccountStateWithProofView, AccountView, CurrencyInfoView, EventView, MetadataView,
         StateProofView, TransactionView,
     },
-    Error, Result,
+    Error, Result, Retry, State,
+};
+use diem_crypto::hash::CryptoHash;
+use diem_types::{
+    account_address::AccountAddress,
+    transaction::{SignedTransaction, Transaction},
 };
-use diem_types::{account_address::AccountAddress, transaction::SignedTransaction};
 use serde::{de::DeserializeOwned, Serialize};
 
 const REQUEST_TIMEOUT: u64 = 10_000;
 
 #[derive(Clone, Debug)]
 pub struct BlockingClient {
     url: String,
+    state: StateManager,
+    retry: Retry,
 }
 
 impl BlockingClient {
     pub fn new<T: Into<String>>(url: T) -> Self {
-        Self { url: url.into() }
+        Self {
+            url: url.into(),
+            state: StateManager::new(),
+            retry: Retry::default(),
+        }
+    }
+
+    pub fn last_known_state(&self) -> Option<State> {
+        self.state.last_known_state()
+    }
+
+    pub fn wait_for_transaction(
+        &self,
+        txn: &SignedTransaction,
+        timeout: Option<Duration>,
+        delay: Option<Duration>,
+    ) -> Result<Response<TransactionView>, WaitForTransactionError> {
+        self.wait_for_transaction_impl(
+            txn.sender(),
+            txn.sequence_number(),
+            txn.expiration_timestamp_secs(),
+            &Transaction::UserTransaction(txn.clone()).hash().to_hex(),
+            timeout,
+            delay,
+        )
+    }
+
+    fn wait_for_transaction_impl(
```

Comment:
we probably should public this method too, for the case only (address, seq, hash) tuple is saved other than the whole SignedTransaction